pkg/microservice/warpdrive/core/service/taskplugin/scanning.go

```diff
@@ -0,0 +1,303 @@
+	p.JobName = jobname
+	p.FileName = filename
+	p.Log = xl
+}
+
+func (p *ScanPlugin) Type() config.TaskType {
+	return p.Name
+}
+
+func (p *ScanPlugin) Status() config.Status {
+	return p.Task.Status
+}
+
+func (p *ScanPlugin) SetStatus(status config.Status) {
+	p.Task.Status = status
+}
+
+func (p *ScanPlugin) TaskTimeout() int {
+	if p.Task.Timeout == 0 {
+		p.Task.Timeout = ScanningTaskTimeout
+	}
+	return int(p.Task.Timeout)
+}
+
+func (p *ScanPlugin) Run(ctx context.Context, pipelineTask *task.Task, pipelineCtx *task.PipelineCtx, serviceName string) {
+	switch p.Task.ClusterID {
+	case setting.LocalClusterID:
+		p.KubeNamespace = zadigconfig.Namespace()
+	default:
+		p.KubeNamespace = setting.AttachedClusterNamespace
+
+		crClient, clientset, restConfig, err := GetK8sClients(pipelineTask.ConfigPayload.HubServerAddr, p.Task.ClusterID)
+		if err != nil {
+			p.Log.Error(err)
+			p.Task.Status = config.StatusFailed
+			p.Task.Error = err.Error()
+			return
+		}
+
+		p.kubeClient = crClient
+		p.clientset = clientset
+		p.restConfig = restConfig
+	}
+
+	// Since only one repository is supported per scanning, we just hard code it
+	repo := &task.Repository{
+		Source:      p.Task.Repos[0].Source,
+		RepoOwner:   p.Task.Repos[0].RepoOwner,
+		RepoName:    p.Task.Repos[0].RepoName,
+		Branch:      p.Task.Repos[0].Branch,
+		PR:          p.Task.Repos[0].PR,
+		OauthToken:  p.Task.Repos[0].OauthToken,
+		Address:     p.Task.Repos[0].Address,
+		Username:    p.Task.Repos[0].Username,
+		Password:    p.Task.Repos[0].Password,
+		EnableProxy: p.Task.Repos[0].EnableProxy,
+		RemoteName:  p.Task.Repos[0].RemoteName,
+	}
+	if repo.RemoteName == "" {
+		repo.RemoteName = "origin"
+	}
+
+	jobCtx := JobCtxBuilder{
+		JobName:     p.JobName,
+		PipelineCtx: pipelineCtx,
+		JobCtx: task.JobCtx{
+			Builds: []*task.Repository{repo},
+		},
+	}
+
+	reaperContext := jobCtx.BuildReaperContext(pipelineTask, serviceName)
+
+	// if the scanning task is of sonar type, then we add the sonar parameter to the context
+	if p.Task.SonarInfo != nil {
+		reaperContext.SonarParameter = p.Task.Parameter
+		reaperContext.SonarServer = p.Task.SonarInfo.ServerAddress
+		reaperContext.SonarLogin = p.Task.SonarInfo.Token
+		reaperContext.ScannerType = ScanningTypeSonar
+	} else {
+		reaperContext.ScannerType = ScanningTypeOther
+		reaperContext.Scripts = append(reaperContext.Scripts, strings.Split(replaceWrapLine(p.Task.Script), "\n")...)
+	}
+
+	jobCtxBytes, err := yaml.Marshal(reaperContext)
+	if err != nil {
+		msg := fmt.Sprintf("cannot reaper.Context data: %v", err)
+		p.Log.Error(msg)
+		p.Task.Status = config.StatusFailed
+		p.Task.Error = msg
+		return
+	}
+
+	jobLabel := &JobLabel{
+		PipelineName: pipelineTask.PipelineName,
+		ServiceName:  serviceName,
+		TaskID:       pipelineTask.TaskID,
+		TaskType:     string(p.Type()),
+		PipelineType: string(pipelineTask.Type),
+	}
+
+	if err := ensureDeleteConfigMap(p.KubeNamespace, jobLabel, p.kubeClient); err != nil {
```

Comment:
If `wd` has been restarted during operation of scan task, the logic here will delete the running task. 
Consider referring to this logic: https://github.com/koderover/zadig/blob/main/pkg/microservice/warpdrive/core/service/taskplugin/build.go#L287-L388?